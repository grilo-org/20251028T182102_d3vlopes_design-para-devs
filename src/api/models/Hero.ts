import { Hero } from '@/api/generated/graphql';

import { ButtonAPI } from './Button';
import { ImageAPI } from './Image';
import { RichTextAPI } from './RichText';

export type HeroAPI = Pick<Hero, 'description'> & {
	title: RichTextAPI;
	bookImage: ImageAPI;
	buttons: ButtonAPI[];
};
