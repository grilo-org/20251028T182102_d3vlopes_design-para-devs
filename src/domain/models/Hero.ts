import { ImageModel } from './Image';

interface ButtonHeroProps {
	text: string;
	target: string;
}

export interface HeroModel {
	heading: string;
	description: string;
	bookImage: ImageModel;
	buttons: {
		primary: ButtonHeroProps;
		secondary: ButtonHeroProps;
	};
}
