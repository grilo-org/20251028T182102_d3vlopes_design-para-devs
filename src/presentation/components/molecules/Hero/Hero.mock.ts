import { HeroProps } from '.';

export const heroMock: HeroProps = {
	heading:
		'<h1>Se torne um desenvolvedor que vai <strong>além do código</strong></h1>',
	description:
		'Aprenda a aplicar conceitos de design no desenvolvimento de software para criar aplicações que vão além de frameworks e libs, elevando a experiência dos usuários.',
	bookImage: {
		src: 'https://media.graphassets.com/NRDgi138ReqCWdYg7l1j',
		alt: 'Capa do eBook com um desenvolvedor em frente ao computador, escrevendo código ',
	},
	buttons: {
		primary: {
			text: 'Comprar agora',
			target:
				'https://pay.hotmart.com/Y88426530P?checkoutMode=10&offDiscount=LANCAMENTO',
		},
		secondary: {
			text: 'Visualizar Preview',
			target: '#preview',
		},
	},
};
