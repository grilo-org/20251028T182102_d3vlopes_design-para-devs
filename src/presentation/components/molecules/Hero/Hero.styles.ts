import { styleWrapper } from '@/libs/styles/tailwind-variants';

const styles = styleWrapper({
	slots: {
		wrapperStyles:
			'relative w-full bg-hero-mobile bg-cover bg-center h-[74.0344rem] pt-[7.0625rem] md:h-[64rem] lg:bg-hero-desktop lg:h-[36.25rem] min-[1520px]:bg-[center_top_-10rem]',
		descriptionStyles:
			'mt-4 font-body text-2xl text-center text-neutral-50 w-[21.4375rem] md:w-[38rem] lg:w-[34rem] lg:text-left xl:w-[46.9375rem]',
		imageStyles:
			'relative z-10 w-[15.3125rem] h-[20.9375rem] mt-10 lg:top-[-80px] lg:left-[-5px] lg:w-[24.1663rem] lg:h-[35.2319rem] lg:mt-0 min-[1100px]:left-[25px] xl:top-[-95px] xl:left-0',
		buttonsWrapperStyles:
			'flex flex-col items-center justify-center gap-8 mt-11 w-full lg:justify-start lg:mt-8 lg:flex-row xl:w-auto',
		buttonPrimaryMobileStyles: 'w-full max-w-[21.4375rem] text-lg',
		buttonPrimaryDesktopStyles: 'w-[12.5rem] text-lg',
		wrapperDesktopStyles:
			'hidden lg:px-[5rem] lg:grid lg:grid-cols-[1fr_1fr] lg:max-w-[78.4287rem] lg:my-0 lg: mx-auto 2xl:px-0',
		wrapperMobileStyles: 'flex flex-col items-center px-4 w-full lg:hidden',
		shapeStyles:
			'hidden lg:block lg:w-full lg:bottom-[-75px] lg:h-[9.5381rem] lg:absolute xl:bottom-[-65px]',
	},
});

const makeStyles = styles();

export const classNames = {
	wrapper: makeStyles.wrapperStyles(),
	description: makeStyles.descriptionStyles(),
	bookImage: makeStyles.imageStyles(),
	buttonsWrapperStyles: makeStyles.buttonsWrapperStyles(),
	buttonPrimaryMobileStyles: makeStyles.buttonPrimaryMobileStyles(),
	buttonPrimaryDesktopStyles: makeStyles.buttonPrimaryDesktopStyles(),
	wrapperDesktopStyles: makeStyles.wrapperDesktopStyles(),
	wrapperMobileStyles: makeStyles.wrapperMobileStyles(),
	shapeStyles: makeStyles.shapeStyles(),
};
