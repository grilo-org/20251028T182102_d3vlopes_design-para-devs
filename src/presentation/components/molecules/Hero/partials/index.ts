export * from './DesktopVersion';
export * from './MobileVersion';
