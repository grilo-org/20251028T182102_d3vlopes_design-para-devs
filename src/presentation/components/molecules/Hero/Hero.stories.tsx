import type { Meta, StoryObj } from '@storybook/react';

import { heroMock } from './Hero.mock';

import { Hero } from '.';

const meta: Meta<typeof Hero> = {
	title: 'Molecules/Hero',
	component: Hero,
	args: heroMock,
	parameters: {
		layout: 'fullscreen',
		design: {
			type: 'figma',
			url: 'https://www.figma.com/file/p1r0o56xHSpG3YHnrNlO3P/LP-Design-para-Devs?type=design&node-id=115-521&mode=design&t=7YVjvjs0AXRkCS68-0',
		},
	},
};

export default meta;
type Story = StoryObj<typeof Hero>;

const template: Story = {
	render: (args) => <Hero {...args} />,
};

export const Default: Story = {
	...template,
};
