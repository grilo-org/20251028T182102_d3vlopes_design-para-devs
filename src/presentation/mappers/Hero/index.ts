import { HeroAPI } from '@/api/models/Hero';
import { HeroModel } from '@/domain/models/Hero';

export class HeroMapper {
	static toDomain(api: HeroAPI): HeroModel {
		return {
			heading: api.title.html.replaceAll('<p>', '').replaceAll('</p>', ''),
			description: api.description,
			bookImage: {
				src: api.bookImage.url,
				alt: String(api.bookImage.imageDescription),
			},
			buttons: {
				primary: {
					text: api.buttons[0].text,
					target: api.buttons[0].target,
				},
				secondary: {
					text: api.buttons[1].text,
					target: api.buttons[1].target,
				},
			},
		};
	}
}
