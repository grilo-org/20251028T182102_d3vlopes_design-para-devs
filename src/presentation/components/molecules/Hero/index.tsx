'use client';

import { HeroModel } from '@/domain/models/Hero';

import { useMediaQuery } from '@/presentation/hooks/useMediaQuery';

import { getTailwindTheme } from '@/utils/getTailwindTheme';

import { classNames } from './Hero.styles';

import { DesktopVersionPartial, MobileVersionPartial } from './partials';

export type HeroProps = HeroModel;

export const Hero = ({ ...props }: HeroProps) => {
	const { screens } = getTailwindTheme();

	const isDesktop = useMediaQuery(screens.lg);

	return (
		<div className={classNames.wrapper}>
			{isDesktop ? (
				<DesktopVersionPartial {...props} />
			) : (
				<MobileVersionPartial {...props} />
			)}
		</div>
	);
};
