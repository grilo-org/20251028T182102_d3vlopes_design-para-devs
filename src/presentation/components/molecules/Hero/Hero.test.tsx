import { render, screen } from '@testing-library/react';

import { heroMock as mock } from './Hero.mock';

import './utils/test/mock';

import { Hero } from '.';

const { getByRole, getAllByRole, getByTestId, getByText } = screen;

describe('<Hero />', () => {
	beforeEach(() => {
		render(<Hero {...mock} />);
	});

	it('should render heading', () => {
		const heading = getByTestId('mock-heading');

		expect(heading).toBeInTheDocument();
	});

	it('should render description', () => {
		const description = getByText(mock.description);

		expect(description).toBeInTheDocument();
	});

	it('should render image', () => {
		const image = getByTestId('mock-image');

		expect(image).toBeInTheDocument();
	});

	it('should render button primary and secondary', () => {
		const [primaryButton, secondaryButton] = getAllByRole('button');

		expect(primaryButton).toBeInTheDocument();
		expect(secondaryButton).toBeInTheDocument();
	});

	it('should render primary button under anchor', () => {
		const primaryAnchor = getByRole('link', {
			name: mock.buttons.primary.text,
		});

		expect(primaryAnchor).toBeInTheDocument();
	});

	it('should render secondary button under anchor', () => {
		const secondaryAnchor = getByRole('link', {
			name: mock.buttons.secondary.text,
		});

		expect(secondaryAnchor).toBeInTheDocument();
	});

	it('should render primary anchor with correct href value', () => {
		const primaryAnchor = getByRole('link', {
			name: mock.buttons.primary.text,
		});

		expect(primaryAnchor).toHaveAttribute('href', mock.buttons.primary.target);
		expect(primaryAnchor).toHaveAttribute('target', '_blank');
	});

	it('should render secondary anchor with correct href value', () => {
		const secondaryAnchor = getByRole('link', {
			name: mock.buttons.secondary.text,
		});

		expect(secondaryAnchor).toHaveAttribute(
			'href',
			mock.buttons.secondary.target,
		);
		expect(secondaryAnchor).not.toHaveAttribute('target', '_blank');
	});

	it('should match snapshot', () => {
		const { container } = render(<Hero {...mock} />);

		expect(container).toMatchSnapshot();
	});
});
