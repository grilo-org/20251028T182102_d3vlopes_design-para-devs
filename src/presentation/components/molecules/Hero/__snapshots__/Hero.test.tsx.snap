// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> should match snapshot 1`] = `
<div>
  <div
    class="relative w-full bg-hero-mobile bg-cover bg-center h-[74.0344rem] pt-[7.0625rem] md:h-[64rem] lg:bg-hero-desktop lg:h-[36.25rem] min-[1520px]:bg-[center_top_-10rem]"
  >
    <div
      class="flex flex-col items-center px-4 w-full lg:hidden"
    >
      <div
        data-testid="mock-heading"
      />
      <p
        class="mt-4 font-body text-2xl text-center text-neutral-50 w-[21.4375rem] md:w-[38rem] lg:w-[34rem] lg:text-left xl:w-[46.9375rem]"
      >
        Aprenda a aplicar conceitos de design no desenvolvimento de software para criar aplicações que vão além de frameworks e libs, elevando a experiência dos usuários.
      </p>
      <div
        class="flex flex-col items-center justify-center gap-8 mt-11 w-full lg:justify-start lg:mt-8 lg:flex-row xl:w-auto"
      >
        <a
          class="w-full max-w-[21.4375rem] text-lg"
          href="https://pay.hotmart.com/Y88426530P?checkoutMode=10&offDiscount=LANCAMENTO"
          rel="noreferrer"
          target="_blank"
        >
          <button
            class="font-heading inline-flex items-center justify-center whitespace-nowrap ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-gradient-primary rounded-[8px] py-6 text-complementary-white font-bold tracking-wider w-full max-w-[21.4375rem] text-lg"
          >
            Comprar agora
          </button>
        </a>
        <a
          href="#preview"
        >
          <button
            class="font-heading inline-flex items-center justify-center whitespace-nowrap ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-transparent text-lg text-neutral-50 md:text-xxl"
          >
            Visualizar Preview
          </button>
        </a>
      </div>
      <div
        class="relative z-10 w-[15.3125rem] h-[20.9375rem] mt-10 lg:top-[-80px] lg:left-[-5px] lg:w-[24.1663rem] lg:h-[35.2319rem] lg:mt-0 min-[1100px]:left-[25px] xl:top-[-95px] xl:left-0"
      >
        <div
          data-testid="mock-image"
        />
      </div>
    </div>
  </div>
</div>
`;
